Feature/azpi arte/nav bar

azpi-arte opened the pull request:
## Tracking Info

Resolves #4 

## Changes

<!-- What changes did you make? -->
- Created semi-responsive navbar with 5 states: Admin, Counselor, Onboarding (Signup), Directory [can temporarily switch between them]
- Created navbar card components for navigation
- Created corresponding navbar pages 
- Set up the layout container for navigation/main content within layout.tsx file
- Set up font and color global variables
- Changed file structure to support Next.js navigation
- Deleted Next.js boilerplate homepage

- TODO
- Need to do onboarding/directory component
- change webpage icon
- set up font SIZE global variables

## Testing

<!-- How did you confirm your changes worked? -->
-Switched between the different navbar state's pages
-Played around with screen size

- TODO
- check if inspector tool gives errors.

## Confirmation of Change

<!-- Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change. -->

![Screenshot 2025-01-18 194326](https://github.com/user-attachments/assets/4a077f51-1293-4342-878e-8420733535d7)
![Screenshot 2025-01-18 194231](https://github.com/user-attachments/assets/6bbb9589-e857-45b1-9ea1-9c624a1345c0)
![Screenshot 2025-01-18 194220](https://github.com/user-attachments/assets/d11e45b4-c607-4ad4-a839-d60f1f4f77cb)
![Screenshot 2025-01-18 194344](https://github.com/user-attachments/assets/a175c745-d84b-4562-a344-99daae178b87)

TODO

Miyuki-L commented on frontend/src/app/components/SideNavbar.tsx:
```diff
@@ -0,0 +1,199 @@
+];
+
+const CounselorItems: CardProps[] = [
+  {
+    iconDark: "dashboard_dark.svg",
+    iconLight: "dashboard_light.svg",
+    navigateTo: "/dashboard",
+    message: "Dashboard",
+  },
```
You can make variables with this 
```suggestion
  Const DashboardProps: CardProps{
    iconDark: "dashboard_dark.svg",
    iconLight: "dashboard_light.svg",
    navigateTo: "/dashboard",
    message: "Dashboard",
  },
```

Then just use the variable in Counselor Items this way you don't need to repeat in the two items.

Miyuki-L commented on frontend/src/app/components/SideNavbar.tsx:
```diff
@@ -0,0 +1,199 @@
+  return (
+    <section className={styles.SideNavbar}>
+      <div className={styles.decoration}>
+        <Image src={logo} alt="" aria-hidden="true" id={styles.logo} />
```
add alt text

Miyuki-L commented on frontend/src/app/components/SideNavbar.tsx:
```diff
@@ -0,0 +1,199 @@
+      <div className={styles.cards}>
+        <span className="text-gray-500">OVERVIEW</span>
+
+        {navState === "Counselor" || navState === "Admin" ? (
```
```
  const renderNavItems = () => {
    switch (navState) {
      case "Counselor":
      case "Admin":
        return items[navState].map((item, index) => (
          <NavCard key={index} {...item} />
        ));
       break;
      case "blank":
        return <div></div>;
      default:
        return null;
    }
  };
```

Try Switch Case Logic. It'll help clear the code up and make it more readable

At the place where you need the nav items (here) `{renderNavItems()}`   should work.

Might need to use `useMemo ` as it depends on the `navState`
DOCS: https://react.dev/reference/react/useMemo

Miyuki-L reviewed (changes requested):
Awesome First task. Needs some clean up and removal of repeated Code
🎊

kirustar14 reviewed (changes requested):
This code seems good, for the lint check it wasn't working for me at first either but I just saved my code, cloned the repository all over again, and made sure to reinstall prettier and the lint thing and it fixed. I'm not sure if you want to try that but it may be an option. And make sure to be in the right directory when trying to run the lint/prettier things. Good job on everything!

azpi-arte commented:
Updated the styling. Made the logo/splagen text responsive. Made logo clickable and navigate to dashboard. Dashboard is now home page. Updated color variables to match figma names. Finished implementing case switching for navbar state

![Screenshot 2025-01-25 141300](https://github.com/user-attachments/assets/5447d284-fc42-4e19-a3e0-b5de174a9b6b)
![Screenshot 2025-01-25 141247](https://github.com/user-attachments/assets/a52ba07d-47f1-4866-bc0e-d407d8cf223a)

Miyuki-L commented on frontend/src/app/components/SideNavbar.module.css:
```diff
@@ -0,0 +1,57 @@
+  align-items: center;
+  margin-bottom: clamp(10px, 5vh, 80px);
+  font-size: 26px;
+  gap: clamp();
```
I believe clamp() needs a parameters are you missing it?

Miyuki-L reviewed (approved):
Other than the comment on the gap clamp Looks good

![image](https://github.com/user-attachments/assets/435e1a41-af8d-40ed-adc2-fd56e55b9517)
Padding around the nav bar when active and hover looks bigger than the figma we can check with designers about it.

Other than that once everyone approves it and changes are fixed you can resolve any conflicts with main and merge it.

kirustar14 reviewed (approved):
Everything seems to look good.
